backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -398,7 +404,21 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
                                 if (it.initializer?.expression !is IrConst<*>?) {
                                     val initialization = evaluateExpression(it.initializer!!.expression)
                                     val address = context.llvmDeclarations.forStaticField(it).storage
-                                    storeAny(initialization, address)
+                                    storeAny(initialization, address, shallFreeze = it.symbol.isShared)
+                                }
+                            }
+                    ret(null)
+                }
+
+                appendingTo(bbLocalInit) {
+                    context.llvm.fileInitializers
+                            .forEach {
+                                if (it.initializer?.expression !is IrConst<*>?) {
+                                   if (!it.symbol.isShared) {
```

Comment:
I believe the same check is required when generating `bbLocalDeinit`.

Reply:
Yep.